sdk/src/main/scala/com/horizen/SidechainMemoryPool.scala

```diff
@@ -109,12 +132,37 @@ class SidechainMemoryPool(unconfirmed: TrieMap[String, SidechainMemoryPoolEntry]
     }
 
     for (t <- txs) {
-      unconfirmed.put(t.id, SidechainMemoryPoolEntry(t))
+      val entry = SidechainMemoryPoolEntry(t)
+      if (entry.feeRate.getFeeRate() >= minFeeRate) {
+        addWithSizeCheck(entry)
+      }
     }
-
     new Success[SidechainMemoryPool](this)
   }
 
+  /**
+   * Add tx to the mempool.
+   * Lowest fee-rate txs are removed in case max mempool size is reached.
+   * @return true if the transaction has been inserted succesfully
+   */
+  def addWithSizeCheck(entry: SidechainMemoryPoolEntry) : Boolean = {
+    val orderedEntries = unconfirmed.values.toSeq.sortWith((a,b) => {
```

Comment:
Just want to point that every time we add new Tx, we sort the whole collection from scratch. It costs.
Can we use some other data type?